### ISSUE TITLE
pgadmin does not respect reverse proxy any more

### ISSUE THREAD

**Issue by fjf2002** [pgadmin-org/pgadmin4#5507]
Hello,

I have an Nginx reverse proxy configuration something like

```nginx
    proxy_set_header X-Script-Name /pgadmin;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header Host $host;
```

... that was working a few months ago (also cf. #4552).

Now it's broken, unfortunately. For example, shortly after clicking the login button, pgadmin calls many URLs **not** containing the `X-Script-Name`, as I can see in Chrome Devtools. For example:

```
GET https://mydomain/static/css/webcabin.overrides.css?v=1667892035310 net::ERR_ABORTED 404 (Not Found)
POST https://mydomain/browser/master_password 404 (Not Found)
GET https://mydomain/browser/check_corrupted_db_file 404 (Not Found)
GET https://mydomain/misc/bgprocess/ 404 (Not Found)
POST https://mydomain/settings/store 404 (Not Found)
POST https://mydomain/misc/cleanup 404 (Not Found)
```

**Comment by kekru** [comment]
Same problem here.

The first wrong request for me is `https://example.com/browser/nodes/`, running with `PGADMIN_CONFIG_SERVER_MODE=False`

According to https://www.pgadmin.org/docs/pgadmin4/latest/container_deployment.html I'm also a little confused about the nginx config. When I look at the Traefik config, it seems that the new path prefix must **not** be stripped away.

That means we should not have slashes after `location` and `proxy_pass`.   
The following sample from the docs wil cause nginx to strip the `/pgadmin4` before forwarding to pgAdmin  

```bash
location /pgadmin4/ {
        proxy_set_header X-Script-Name /pgadmin4;
        proxy_set_header X-Scheme $scheme;
        proxy_set_header Host $host;
        proxy_pass http://localhost:5050/;
        proxy_redirect off;
}
```

I think it should be 

```bash
location /pgadmin4 {
        proxy_set_header X-Script-Name /pgadmin4;
        proxy_set_header X-Scheme $scheme;
        proxy_set_header Host $host;
        proxy_pass http://localhost:5050;
        proxy_redirect off;
}
```

**Comment by bigteejay** [comment]
Upon my trying the same, I am presented with the expected login screen.

After successful login (no error message), rather than being redirected to something like the `X-Script-Name` path (eg: `/subdir1`) and a usable interface, I am redirected to the web/server root (`/`, which in my `nginx.conf`, there is nothing for that route).

If I try going back to `/subdir1` I am then redirected to a logged in `subdir1/browser/` and the site/app is usable.  How can the redirect ignoring `X-Script-Name` after login be solved?

**Comment by bigteejay** [comment]
This may be related to a [fix](https://github.com/pgadmin-org/pgadmin4/commit/4d9dfe7d041be8730d7beda94311566d7267cb33) [introduced](https://github.com/pgadmin-org/pgadmin4/issues/5343) in 6.14?

**Comment by akshay-joshi** [comment]
Fixed in PR https://github.com/pgadmin-org/pgadmin4/pull/5523

**Comment by yogeshmahajan-1903** [comment]
This is working fine . Verified with Desktop & Server mode configured with nginx reverse proxy.

**Comment by fjf2002** [comment]
I can confirm: the bug is solved in 6.16.
